src/CodeGen/GenerateSyntax.hs

```diff
@@ -108,7 +138,7 @@ addTickIfNecessary s
 
 -- | Convert snake_case string to CamelCase Name
 toName :: String -> Name
-toName = mkName . toCamelCase
+toName = mkName . addTickIfNecessary . toCamelCase
```

Comment:
I don’t think we need to do `addTickIfNecessary` here, since `toCamelCase` does `initUpper` and thus the name won’t ever be one of the keywords (`type`, `module`, `data`) that `addTickIfNecessary` was intended to deal with.

By contrast, we use `addTickIfNecessary` in `toVarBangType`, and that seems correct to me.

Reply:
Aha. @aymannadeem asked why we didn't use `addTickIfNecessary` there, and I could not come up with a good answer.

Reply:
This might indicate an opportunity for a better name ☝️

Reply:
I intend to handle this and the refactoring related to consolidating `toName` and `toName'` on a separate branch.